src/io/flutter/coverage/ScriptManager.java

```diff
@@ -0,0 +1,191 @@
+              return library.getScripts().get(0);
+            }
+          }
+        }
+      }
+    }
+
+    return null;
+  }
+
+  @Nullable
+  private Library getLibrary(LibraryRef libraryRef) {
+    final Ref<Library> resultRef = Ref.create();
+    final Semaphore semaphore = new Semaphore();
+    semaphore.down();
+
+    vmService.getLibrary(isolateRef.getId(), libraryRef.getId(), new GetLibraryConsumer() {
+      @Override
+      public void received(Library library) {
+        resultRef.set(library);
+        semaphore.up();
+      }
+
+      @Override
+      public void onError(RPCError error) {
+        semaphore.up();
+      }
+    });
+    semaphore.waitFor(RESPONSE_WAIT_TIMEOUT);
+    return resultRef.get();
+  }
+
+  public void populateFor(ScriptRef scriptRef) {
+    if (!scriptMap.containsKey(scriptRef.getId())) {
+      scriptMap.put(scriptRef.getId(), getScriptSync(scriptRef));
+      linesAndColumnsMap.put(scriptRef.getId(), createTokenPosToLineAndColumnMap(scriptMap.get(scriptRef.getId())));
+    }
+  }
+
+  public Pair<Integer, Integer> getLineColumnPosForTokenPos(ScriptRef scriptRef, int tokenPos) {
+    final TIntObjectHashMap<Pair<Integer, Integer>> map = linesAndColumnsMap.get(scriptRef.getId());
+    return map == null ? null : map.get(tokenPos);
+  }
+
+  private Script getScriptSync(@NotNull final ScriptRef scriptRef) {
+    final Ref<Script> resultRef = Ref.create();
+    final Semaphore semaphore = new Semaphore();
+    semaphore.down();
+
+    vmService.getObject(isolateRef.getId(), scriptRef.getId(), new GetObjectConsumer() {
+      @Override
+      public void received(Obj script) {
+        resultRef.set((Script)script);
+        semaphore.up();
+      }
+
+      @Override
+      public void received(Sentinel response) {
+        semaphore.up();
+      }
+
+      @Override
+      public void onError(RPCError error) {
+        semaphore.up();
+      }
+    });
+
+    semaphore.waitFor(RESPONSE_WAIT_TIMEOUT);
+    return resultRef.get();
+  }
+
+  private static TIntObjectHashMap<Pair<Integer, Integer>> createTokenPosToLineAndColumnMap(@Nullable final Script script) {
+    if (script == null) {
+      return null;
+    }
+
+    // Each subarray consists of a line number followed by (tokenPos, columnNumber) pairs;
+    // see https://github.com/dart-lang/vm_service_drivers/blob/master/dart/tool/service.md#script.
+    final TIntObjectHashMap<Pair<Integer, Integer>> result = new TIntObjectHashMap<>();
+
+    for (List<Integer> lineAndPairs : script.getTokenPosTable()) {
+      final Iterator<Integer> iterator = lineAndPairs.iterator();
+      final int line = Math.max(0, iterator.next() - 1);
+      while (iterator.hasNext()) {
+        final int tokenPos = iterator.next();
+        final int column = Math.max(0, iterator.next() - 1);
+        result.put(tokenPos, Pair.create(line, column));
+      }
+    }
+
+    return result;
+  }
+
+  public Script getScriptFor(ScriptRef ref) {
```

Comment:
Might `ref` be null?

Reply:
I changed the param to be NotNull (and the return to be Nullable).